meteor/server/api/ingest/shouldRemoveOrphanedPartInstance.ts

```diff
@@ -0,0 +1,88 @@
+import {
+	BlueprintRemoveOrphanedPartInstance,
+	ShowStyleBlueprintManifest,
+} from '@sofie-automation/blueprints-integration'
+import { ReadonlyDeep } from 'type-fest'
+import { PartNote, SegmentNote } from '../../../lib/api/notes'
+import { Rundown } from '../../../lib/collections/Rundowns'
+import { ShowStyleCompound } from '../../../lib/collections/ShowStyleVariants'
+import { clone, literal, unprotectObject, unprotectObjectArray } from '../../../lib/lib'
+import { logger } from '../../logging'
+import { RemoveOrphanedPartInstanceContext } from '../blueprints/context/removeOrphanedPartInstance'
+import { CacheForPlayout, getSelectedPartInstancesFromCache } from '../playout/cache'
+import { isTooCloseToAutonext } from '../playout/lib'
+
+export async function shouldRemoveOrphanedPartInstance(
+	cache: CacheForPlayout,
+	showStyle: ReadonlyDeep<ShowStyleCompound>,
+	blueprint: ReadonlyDeep<ShowStyleBlueprintManifest>,
+	rundown: ReadonlyDeep<Rundown>
+): Promise<void> {
+	if (!cache.Playlist.doc.activationId) return
+	if (!blueprint.shouldRemoveOrphanedPartInstance) return
+
+	const playlistPartInstances = getSelectedPartInstancesFromCache(cache)
+	if (!playlistPartInstances.nextPartInstance?.orphaned) return
```

Comment:
Are adlibbed instances taking the correct route here?
I think they ought to be caught in this return too

Reply:
If `orphaned` is set then it's either deleted or adLibbed, so this should be correct

Reply:
I dont know what my logic behind this comment was...